eng/emitter-package.json

```diff
@@ -0,0 +1,18 @@
+{
+  "main": "dist/src/index.js",
+  "dependencies": {
+    "@azure-tools/typespec-go": "0.1.0"
+  },
+  "devDependencies": {
+    "@azure-tools/typespec-autorest": "0.44.0",
```

Comment:
Go emitter was not moved to TypeSpec 0.58 yet.  Please hold the merge until Emitter update has completed and released.